.github/workflows/tracegen.yml

```diff
@@ -0,0 +1,60 @@
+name: tracegen
+on:
+  push:
+    branches: [ main ]
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+*'
+  pull_request:
+
+jobs:
+  build-dev:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v2
+      - name: Build tracegen
+        run: |
+          docker build tracegen -t ghcr.io/open-telemetry/opentelemetry-collector-contrib/tracegen:dev
+
+  publish-dev:
+    runs-on: ubuntu-latest
+    if: (github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags/v')) && github.repository == 'open-telemetry/opentelemetry-collector-contrib'
+    permissions:
+      packages: write
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v2
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Push tracegen to Github packages
+        run: |
+          docker build tracegen -t ghcr.io/open-telemetry/opentelemetry-collector-contrib/tracegen:latest
+          docker push ghcr.io/open-telemetry/opentelemetry-collector-contrib/tracegen:latest
+
+  publish-stable:
+    runs-on: ubuntu-latest
+    if: startsWith(github.ref, 'refs/tags/v') && github.repository == 'open-telemetry/opentelemetry-collector-contrib'
+    permissions:
+      packages: write
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v2
+      - name: Set Release Tag
+        id: github_tag
+        run: ./.github/workflows/scripts/set_release_tag.sh
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Push tracegen to Github packages
+        run: |
+          docker build tracegen -t ghcr.io/open-telemetry/opentelemetry-collector-contrib/tracegen:$RELEASE_TAG
+          docker push ghcr.io/open-telemetry/opentelemetry-collector-contrib/tracegen:$RELEASE_TAG
```

Comment:
There's also an action for pushing the image: https://github.com/docker/build-push-action

Reply:
fixed.

Reply:
I'm still using the docker command here cuz we extracted `RELEASE_TAG` from scripts. It will help us to keep image tags consistent and avoid future compatibility issues. I checked metadata-action and the rule is a little bit different with currently we have.



Event | Ref | Docker Tags
-- | -- | --
pull_request | refs/pull/2/merge | pr-2
push | refs/heads/master | master
push | refs/heads/releases/v1 | releases-v1
push tag | refs/tags/v1.2.3 | v1.2.3, latest
push tag | refs/tags/v2.0.8-beta.67 | v2.0.8-beta.67, latest